packages/doenetml-worker-rust/lib-doenetml-core/src/components/doenet/p.rs

```diff
@@ -1,59 +1,37 @@
-use std::collections::HashMap;
-use strum::VariantNames;
-
 use crate::components::prelude::*;
+use crate::general_prop::RenderedChildrenPassthroughProp;
 
-use crate::general_prop::BooleanProp;
-
-#[derive(Debug, AttributeProp)]
-pub enum PAttribute {
-    /// Whether the `<p>` should be hidden.
-    #[attribute(prop = BooleanProp, default = false)]
-    Hide,
-}
-
-#[derive(Debug, Default, ComponentNode, ComponentChildrenOld, ComponentProps, ComponentActions)]
-#[pass_through_children]
-pub struct P {
-    pub common: ComponentCommonData,
-
-    pub props: PProps,
-}
+/// The `<p>` component renders its children
+#[component(name = P, rendered_children = "passthrough")]
+mod component {
 
-#[derive(Debug, ComponentProps)]
-pub struct PProps {
-    /// A variable that determines whether or not a p should be sent to the renderer (i.e., appear in the render tree).
-    ///
-    /// If `hidden` is true, then don't send the p to the renderer. (TODO: implement this)
-    ///
-    /// It is marked `is_public` so that it can be referenced in DoenetML via `.hidden`.
-    #[is_public]
-    hidden: Prop<bool>,
-}
+    use crate::general_prop::BooleanProp;
 
-impl PProps {
-    fn new() -> Self {
-        PProps {
-            hidden: PAttribute::Hide.prop(),
-        }
+    enum Props {
+        /// Whether the `<p>` should be hidden.
+        #[prop(value_type = PropValueType::Boolean, profile = PropProfile::Hidden)]
+        Hidden,
+        #[prop(value_type = PropValueType::GraphNodes, profile = PropProfile::RenderedChildren)]
+        RenderedChildren,
     }
-}
 
-impl Default for PProps {
-    fn default() -> Self {
-        PProps::new()
+    enum Attributes {
+        /// Whether the `<p>` should be hidden.
+        #[attribute(prop = BooleanProp, default = false)]
+        Hide,
     }
 }
 
-impl ComponentAttributes for P {
-    fn get_attribute_names(&self) -> Vec<AttributeName> {
-        PAttribute::VARIANTS.into()
-    }
-}
+pub use component::PActions;
+pub use component::PAttributes;
+pub use component::PProps;
+pub use component::P;
 
-impl ComponentChildren for P {
-    fn get_rendered_children(&self, child_query_object: ChildQueryObject) -> Vec<GraphNode> {
-        // Return children without modification
-        child_query_object.child_iter().collect()
+impl PropGetUpdater for PProps {
+    fn get_updater(&self) -> Box<dyn PropUpdater> {
+        match self {
+            PProps::Hidden => PAttributes::Hide.get_prop_updater(),
+            PProps::RenderedChildren => Box::new(RenderedChildrenPassthroughProp::new()),
```

Comment:
Since we need updaters more often than anything else, I think we should add a `new_updater()` function that already returns a boxed type.

Reply:
Would it be bad if all the regular `new_xxx()` functions always returned the boxed type? We never need anything other than the boxed version. Not sure how adding a typed layer would change the interface.

Reply:
I think that would be fine. Or maybe `new_boxed()`?